Program.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+internal class Program
+{
+    static void Main(string[] args)
+    {
+        Console.WriteLine("введите значение х");
+        double x = double.Parse(Console.ReadLine());
+        static double vicheslenie(ref double x)
+        {
+            if(Math.Sin(x)<-0.5)
+            {
+                return Math.Sin(x);
+            }
+            else if ( -0.5 >= Math.Sin(x) & Math.Sin(x) <= 0.5)
+            {
+                return 0;
+            }
+            else
+            {
+                return 1;
+            }
+        }
+        Console.WriteLine(vicheslenie(ref x));
+    }

```

Comment:
1.  Без особых причин лучше не использовать вложенные методы.
2. Именуйте метод с большой буквы и желательно начиная с глагола. Не используйте в названии метода транслитерацию.
3. Без особых причин не передавайте аргументы в метод по ссылке.
4. Если внутри `if` только одна строка, то `{ }` можно не писать.
5. Если внутри `if` стоит `return`, то дальнейший `else` излишен.
6. Можно не вычислять одну и ту же величину несколько раз, а запомнить её значение в локальной переменной.
7. Избегайте очевидно лишних проверок.
```suggestion
    static void Main()
    {
        Console.WriteLine("Введите значение х");
        double x = double.Parse(Console.ReadLine());
        
        Console.WriteLine(Calculate(x));
    }
    
    static double Calculate(double x)
    {
        double s = Math.Sin(x);
        
        if(s < -0.5)
            return s;

        if ( s <= 0.5)
                return 0;
                
        return 1;
     }
```